scala-api/src/main/scala/com/cloudera/livy/package.scala

```diff
@@ -17,67 +17,52 @@
 
 package com.cloudera.livy
 
-import java.io.File
-import java.net.URI
+import org.apache.spark.SparkContext
+import org.apache.spark.sql.SQLContext
+import org.apache.spark.streaming.StreamingContext
 
-import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Promise
 import scala.concurrent.Future
 
 package object client {
 
-  case class LivyScalaClient(livyJavaClient: LivyClient) {
-
-    def submit[T](block: JobContext => T): JobHandle[T] = {
-
-      val job = new Job[T] {
-        @throws(classOf[Exception])
-        override def call(jobContext: JobContext): T = block(jobContext)
-      }
-
-      livyJavaClient.submit(job)
-    }
+  implicit class ScalaWrapper(livyJavaClient: LivyClient) {
+    def asScalaClient = new LivyScalaClient(livyJavaClient)
+  }
 
-    def run[T](block: JobContext => T): Future[_] = {
+  def asScalaJobContext(context: JobContext): ScalaJobContext = {
```

Comment:
Having a method like this is not necessary. Declare `ScalaJobContext` in its own file, and just create it by, well, creating an instance (`new ScalaJobContext(javaJobContext)`).